homeassistant/components/slack/notify.py

```diff
@@ -61,6 +93,26 @@ async def async_get_service(hass, config, discovery_info=None):
     )
 
 
+@callback
+def _async_get_filename_from_url(url):
+    """Return the filename of a passed URL."""
+    parsed_url = urlparse(url)
+    return os.path.basename(parsed_url.path)
+
+
+async def _async_get_remote_file_contents(hass, url, *, username=None, password=None):
+    """Get the binary contents of a remote file."""
+    session = aiohttp_client.async_get_clientsession(hass)
+
+    kwargs = {}
+    if username and password:
```

Comment:
You should check if `password` is not None`, or else you can't provide an empty password.